### ISSUE TITLE
DocCommentSniff reporting single-line block comments as error

### ISSUE THREAD

**Issue by aik099** [squizlabs/PHP_CodeSniffer#258]
There are 2 way how you can write a block comment:

``` php
class Something
{
    /**
     * Description.
     *
     * @var ClassName
     */
    protected $somethingMultiLine;

    /** @var ClassName Description */
    protected $somethingElse;

}
```

Same goes for method DocBlocks and DocBlocks for individual variables in the code.

However the `DocCommentSniff` reports 1st line in following code as error:

``` php
/** @var NodeElement $content_box_title */
$content_box_title = array_shift($content_box_titles);
```

Error text: `The open comment tag must be the only content on the line`
Error code: `ContentAfterOpen`

Reference: http://www.phpdoc.org/docs/latest/references/phpdoc/tags/var.html

**Comment by aik099** [comment]
ping

**Comment by TomasVotruba** [comment]
What's the current status of this? I think it didn't came across this using single line docs.

**Comment by aik099** [comment]
The variable single line docs is common use case, when you want to have IDE method/property autocomplete, but object in a variable is created by service container method, that have same return type for all objects.

**Comment by TomasVotruba** [comment]
Ah, I probably don't use this Sniff then. Never came across this issue.

**Comment by vedmant** [comment]
Just came across this issue as well, it looks like I need either to disable doc comments sniff or avoid using shorthand @var declarations, both variants are not appealing to me.

**Comment by miken32** [comment]
So, what is the argument for not including this in the generic checks? Given that PHPDoc itself is a mature standard, I don't see any reason to wait for PSR-5, if it's even completed; it will just refine the existing PHPDoc standard.

**Comment by oojacoboo** [comment]
Bump 👍

**Comment by mikaelz** [comment]
Valid
```
/**
 * Import count
 *
 * @var int
 */
private $import_count = 0;
```
Invalid :(
```
/** @var int $import_count Import count */
private $import_count = 0;

```

**Comment by LastRide** [comment]
Ping.

**Comment by glen-84** [comment]
Is the fact that:

```php
/* @var $matches array */
```

(note single asterisk)

... results in:

`Single line block comment not allowed; use inline ("// text") comment instead`

... related to this issue?

It would be nice to allow the above while still disallowing single-line block comments that don't include the `@var`.

**Comment by gsherwood** [comment]
> ... related to this issue?

I don't think so. Only comments that start with `/**` are tokenized as doc blocks. Any comments that just start with `/*` are considered inline comments and are picked up by a different sniff banning that syntax for single-line inline comments.

I'm assuming you are documenting variables inside code and not class properties? If so, there are no sniffs or standards included with PHPCS to enforce any rules in this area, so you're probably asking for a different new feature.

**Comment by glen-84** [comment]
> I'm assuming you are documenting variables inside code and not class properties?

Correct. I've opened #1977.

**Comment by VincentLanglet** [comment]
What is the status of this ?

**Comment by larssn** [comment]
@VincentLanglet It's still not supported, and I assume this answer https://github.com/squizlabs/PHP_CodeSniffer/pull/276#issuecomment-58981916 is the official stance on the matter.

I'm disappoint

**Comment by bkdotcom** [comment]
Started using static analysis tool psalm  [which uses inline /** @var */ ](https://psalm.dev/docs/annotating_code/typing_in_psalm/)
trying to figure out how to allow this use-case...

**Comment by glen-84** [comment]
@bkdotcom You can do something like [this](https://github.com/glen-84/phpcs-standard/blob/52b5eeac9bb99fd1e0c6aeb726baf487910c284b/Glen84/ruleset.xml#L116-L118) for now.

**Comment by bkdotcom** [comment]
@glen-84  thanks..  I also had to exclude
Squiz.Commenting.BlockComment.WrongStart
and
Squiz.Commenting.InlineComment.DocBlock

I believe the only relevant sniffs left in play are the PSR12 sniffs
